src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java

```diff
@@ -0,0 +1,107 @@
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class PerformanceAnalyzerIT extends ESRestTestCase {
+    private static final Logger LOG = LogManager.getLogger(PerformanceAnalyzerIT.class);
+    private static final int PORT = 9600;
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static RestClient paClient;
+
+    @Before
+    public void initPaClient() throws IOException {
+        String cluster = System.getProperty("tests.rest.cluster");
+        logger.info("Cluster is {}", cluster);
+        if (cluster == null) {
+            throw new RuntimeException("Must specify [tests.rest.cluster] system property with a comma delimited list of [host:port] "
+                    + "to which to send REST requests");
+        }
+        List<HttpHost> hosts = Collections.singletonList(
+                buildHttpHost(cluster.substring(0, cluster.lastIndexOf(":")), PORT));
+        logger.info("initializing PerformanceAnalyzer client against {}", hosts);
+        paClient = buildClient(restClientSettings(), hosts.toArray(new HttpHost[0]));
+    }
+
+
+    public static void ensurePaAndRcaEnabled() throws Exception {
+        // TODO replace with waitFor with a 1min timeout
+        for (int i = 0; i < 60; i++) {
+            Response resp = client().performRequest(new Request("GET", "_opendistro/_performanceanalyzer/cluster/config"));
+            Map<String, Object> respMap = mapper.readValue(EntityUtils.toString(resp.getEntity(), "UTF-8"),
+                    new TypeReference<Map<String, Object>>(){});
+            if (respMap.get("currentPerformanceAnalyzerClusterState").equals(3)) {
+                break;
+            }
+            Thread.sleep(1000L);
+        }
+        Response resp = client().performRequest(new Request("GET", "_opendistro/_performanceanalyzer/cluster/config"));
+        Map<String, Object> respMap = mapper.readValue(EntityUtils.toString(resp.getEntity(), "UTF-8"),
+                new TypeReference<Map<String, Object>>(){});
+        if (!respMap.get("currentPerformanceAnalyzerClusterState").equals(3)) {
+            throw new Exception("PA and RCA are not enabled on the target cluster!");
+        }
+    }
+
+    private static class TestUtils {
+        public static final String DATA = "data";
+        public static final String RECORDS = "records";
+
+        // Field related strings
+        public static final String FIELDS = "fields";
+        public static final String FIELD_NAME = "name";
+        public static final String FIELD_TYPE = "type";
+        public static final String DOUBLE_TYPE = "DOUBLE";
+
+        // Metrics related strings
+        public static final String M_DISKUTIL = "Disk_Utilization";
+    }
```

Comment:
nit: move the class definition to the end of the class.

Reply:
Done